src/DynamoCore/Core/Threading/DynamoScheduler.cs

```diff
@@ -74,6 +74,18 @@ public interface IScheduler
         IEnumerable<AsyncTask> Tasks { get; }
```

Comment:
Thanks! What I meant was... returning a list of `AsyncTask` objects to the caller, even with lock, is not thread-safe, the lock does not live beyond the property-access. The caller can then gain access to these tasks that might be used in a separate thread. I'm suggesting that this property be removed if it doesn't serve any purposes other than to do an `Any()`. If there's a reason for caller to access the tasks themselves, then we might want to know why and see if that can be tweaked.